experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -126,7 +134,12 @@ export class NodeSDK {
 
     this._autoDetectResources = configuration.autoDetectResources ?? true;
 
-    if (configuration.spanProcessor || configuration.traceExporter) {
+    // If there are configuration options to read, we need to create a new TracerProvider even if the exporter is configured in the environment
+    if (
+      configuration.traceExporter ||
+      configuration.spanProcessor ||
+      envExporterWithConfiguration
```

Comment:
Span processor creation at L156-157 was relying on that configuration containing having either of these:
- a span processor
- an exporter

Since there's a third case now (no actual exporter set, only the environment variable, and no span processor set), it instantiates a BatchSpanProcessor with an undefined SpanExporter there, which causes two span processors to be registered.

Reply:
I see. I've updated the logic now such that instead of relying on updating the if case here, we just keep track of the configuration options passed to the SDK at initialization time so that if the user configures the SDK with an env exporter and manual sampler we configure the `tracerProvider` created on line 328 appropriately with the configuration options they selected, while still only creating one span processor of the `TracerProviderWithEnvExporters` type. I also added a test to cover this case.